### PR TITLE
refactor(ui): migrate machine list tests to typescript

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Migrate the test files for some machine list components to typescript.

## QA

N/A

## Fixes

Fixes: #2717.

### CODE REVIEW

File: ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx [hunk 1]
```diff
@@ -240,50 +242,4 @@ describe("MachineListHeader", () => {
       wrapper.find('Button[data-test="add-hardware-dropdown"]').length
     ).toBe(0);
   });
-
-  it("can add the selected filter when displaying a form", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx:244]: Is this going to get covered in the selected actions update?

File: ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx [hunk 2]
```diff
@@ -240,50 +242,4 @@ describe("MachineListHeader", () => {
...
-    act(() =>
-      wrapper
-        .find("TakeActionMenu")
-        .props()
-        .setSelectedAction({ name: NodeActions.SET_POOL })
-    );
-    expect(setSearchFilter).toHaveBeenCalledWith("in:(selected)");
-  });
-
-  it("can remove the selected filter when cancelling a form", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx:268]: Same as above

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Just a couple of questions on a couple of tests which I assume will be added as part of the selected action changes

File: ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx:244]: Yeah I missed that I'd removed those ones, I've got them in a different component in my other branch.

File: ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx:268]: Same as above

**Pull request merged**